File: test_source_code_examples/java/ComplexData.java
import java.util.List;
import java.util.Map;

public class ComplexData {
    private byte[] raw_bytes;
    private List<String> items;
    private Map<String, Integer> config;
    private User owner;

    public void process(String mode) {
        // ...
    }

    public static ComplexData create() {
        return new ComplexData();
    }

    private class Metadata {
        public String key;
        public String value;

        public void log() {}
    }
}

File: languages/objc/examples/ComplexData.m
#import <Foundation/Foundation.h>

@class User;

@interface Metadata : NSObject
@property (nonatomic, copy) NSString *key;
@property (nonatomic, copy) NSString *value;
- (void)log;
@end

@implementation Metadata
- (void)log {}
@end

@interface ComplexData : NSObject
@property (nonatomic, strong) NSData *raw_bytes;
@property (nonatomic, strong) NSArray<NSString *> *items;
@property (nonatomic, strong) NSDictionary<NSString *, NSNumber *> *config;
@property (nonatomic, strong) User *owner;

- (void)process:(NSString *)mode;
+ (ComplexData *)create;
@end

@implementation ComplexData
- (void)process:(NSString *)mode {
    // ...
}
+ (ComplexData *)create {
    return [[ComplexData alloc] init];
}
@end
